matrix/helper_functions.hpp

```diff
@@ -20,64 +20,40 @@ bool is_finite(Type x) {
 #endif
 }
 
+/**
+ * Wrap value to stay in range [low, high)
+ *
+ * @param x input possibly outside of the range
+ * @param low lower limit of the allowed range
+ * @param high upper limit of the allowed range
+ * @return wrapped value inside the range
+ */
 template<typename Type>
-Type wrap_pi(Type x)
-{
-    if (!is_finite(x)) {
-        return x;
-    }
-
-    int c = 0;
-
-    while (x >= Type(M_PI)) {
-        x -= Type(2 * M_PI);
-
-        if (c++ > 100) {
-            return INFINITY;
-        }
-    }
-
-    c = 0;
-
-    while (x < Type(-M_PI)) {
-        x += Type(2 * M_PI);
-
-        if (c++ > 100) {
-            return INFINITY;
-        }
+Type wrap(Type x, Type low, Type high) {
+    Type range = high - low;
+    x = fmod(x - low, range);
```

Comment:
Would `fmodf` be more appropriate here?

Reply:
oops - i missed the discussion in the issue. looks like it was already talked about and benchmarked. dont mind me :)